presto-main-base/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -2372,12 +2372,26 @@ private Scope processView(Table table, Optional<Scope> scope, QualifiedObjectNam
 
             analysis.getAccessControlReferences().addViewDefinitionReference(name, view);
 
+            Optional<Expression> savedViewAccessorWhereClause = Optional.empty();
+            if (analysis.getCurrentQuerySpecification().isPresent()) {
+                QuerySpecification currentQuerySpec = analysis.getCurrentQuerySpecification().get();
+                savedViewAccessorWhereClause = currentQuerySpec.getWhere();
+                if (savedViewAccessorWhereClause.isPresent()) {
+                    analysis.setViewAccessorWhereClause(savedViewAccessorWhereClause);
+                }
+            }
```

Comment:
```suggestion
            Optional<Expression> savedViewAccessorWhereClause = analysis.getCurrentQuerySpecification().flatMap(QuerySpecification::getWhere);
            if (savedViewAccessorWhereClause.isPresent()) {
                analysis.setViewAccessorWhereClause(savedViewAccessorWhereClause);
            }
```

nit: consider using a more functional style here